rasa/core/training/story_conflict.py

```diff
@@ -276,10 +330,30 @@ def _sliced_states_iterator(
                 sliced_states = MaxHistoryTrackerFeaturizer.slice_state_history(
                     states[: idx + 1], max_history
                 )
+                if tokenizer:
+                    _apply_tokenizer_to_states(tokenizer, sliced_states)
+                # ToDo: deal with oov (different tokens can lead to identical features if some of those tokens are out of vocabulary for all featurizers)
                 yield TrackerEventStateTuple(tracker, event, sliced_states)
                 idx += 1
 
 
+def _apply_tokenizer_to_states(tokenizer: Tokenizer, states: List[State]) -> None:
+    """Split each user text into tokens and concatenate them again.
+
+    Args:
+        tokenizer: A tokenizer to tokenize the user messages.
+        states: The states to be tokenized.
+    """
+    for state in states:
+        if USER in state:
+            state[USER][TEXT] = "".join(
```

Comment:
don't we want to keep at least the spaces?
```suggestion
            state[USER][TEXT] = " ".join(
```

Reply:
I don't think there is any point in doing so as this is only to create the hash, and fewer characters means faster hash.

Reply:
I'll add the space anyway.